### PR TITLE
LESQ-1004  - style: onboarding -  do you work in a school form

### PR DESCRIPTION
**Pull request by JBR90**
## Description

Music year: 1991

- Add's AVO page name and URL updates for new onboarding pages
- Refactors 'do you work in a school' onboarding view to use shared onboarding form.
- Correct validation and error for continue
- Onboarding form to have optional newsletter and terms and conditions
- Refactors shared onboarding form to use field-set and legend instead of heading

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2700--oak-web-application.netlify.thenational.academy/onboarding

## Acceptance Criteria:

- [ ]  First page that is shown after Auth0, regardless of location
- [ ]  Yes takes you on the school picker journey
- [ ]  No takes you on the “who the hell are you then?” journey
- [ ]  Nothing selected by default, if a user selects “Continue” without selecting an option, error!
    - “Please select if you work in a school”
    - Error is associated with input field

## Screenshots
There is extra spacing above the error due to how we have implemented the shared onboarding form. This could be addressed in the shared form but may have a knock on effect on the style of the other forms
image
## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

File: src/components/TeacherViews/Onboarding/ONboarding.test.tsx [hunk 1]
```diff
@@ -1,16 +1,60 @@
 import { screen } from "@testing-library/dom";
+import userEvent from "@testing-library/user-event";
+import { act } from "@testing-library/react";
```
Comment (k-huggs) [inline src/components/TeacherViews/Onboarding/ONboarding.test.tsx:4]: Is there a reason the test file is called `ONboarding`?

File: src/components/TeacherViews/Onboarding/ONboarding.test.tsx [hunk 2]
```diff
@@ -1,16 +1,60 @@
...
+    await act(async () => {
+      await userEvent.click(screen.getByRole("radio", { name: "Yes" }));
+      await userEvent.click(screen.getByText("Continue"));
+    });
+
+    const error = screen.queryByText(/Please select if you work in a school/i);
+
+    expect(error).not.toBeInTheDocument();
+  });
+  it("does not show the error message initially", () => {
```
Comment (k-huggs) [inline src/components/TeacherViews/Onboarding/ONboarding.test.tsx:54]: would it be worth adding to test that the error message is displayed?

Comment (JBR90) [reply src/components/TeacherViews/Onboarding/ONboarding.test.tsx:54]: Yeah not sure how i missed that!

File: src/components/TeacherViews/Onboarding/ONboarding.test.tsx (hunk 1 above)
Comment (JBR90) [reply src/components/TeacherViews/Onboarding/ONboarding.test.tsx:4]: Yeah i'm not sure, was like that after merging in main, it wont change back.

**Review by k-huggs — APPROVED** [review]

File: src/components/TeacherViews/Onboarding/Onboarding.view.tsx [hunk 3]
```diff
@@ -45,79 +33,39 @@ export const OnboardingView = () => {
...
-    <OakThemeProvider theme={oakDefaultTheme}>
-      <OakMaxWidth>
-        <OakFlex
-          $flexDirection="column"
-          $width="all-spacing-21"
-          $gap="space-between-m"
+    <OakFlex $background={"bg-decorative1-main"}>
+      <OakMaxWidth $justifyContent={"center"} $height={"100vh"}>
+        <OnboardingForm
+          control={control as Control<OnboardingFormProps>}
```
[the remark this answers is not in the document]
Comment (JBR90) [reply src/components/TeacherViews/Onboarding/Onboarding.view.tsx:40]: This was from @njblore PR, there was some effort to make this generic but was successfull due to react-form

**Review by mikeritson-oak — APPROVED** [review]
Work in this PR checked and passed. Form is present, functional and directs users to the relevant next step, with or without error messages.

**Pull request merged**